src/js/components/search/table/ResultsTable.jsx

```diff
@@ -41,9 +41,11 @@ export default class ResultsTable extends React.PureComponent {
     }
 
     componentWillReceiveProps(nextProps) {
+        const currentType = nextProps.currentType;
+        const visibileCols = parseFloat(nextProps.columns.length);
```

Comment:
1. `visible` is spelled incorrectly.
2. Is getting a count of the column length sufficient to generate a hash with? It may be the case that the user removes an existing column and adds a new one - new data but the count is different. Many ImmutableJS objects can give you [hashCodes](https://facebook.github.io/immutable-js/docs/#/OrderedSet/hashCode) to check for equality that may be useful for this.